pkg/operator/staticpod/controller/node/node_controller.go

```diff
@@ -86,14 +86,26 @@ func (c NodeController) sync(ctx context.Context, syncCtx factory.SyncContext) e
 	}
 
 	// detect and report master nodes that are not ready
+	nodeConditionFinder := func(node *coreapiv1.Node) *coreapiv1.NodeCondition {
+		for i := range node.Status.Conditions {
+			if node.Status.Conditions[i].Type == coreapiv1.NodeReady {
+				return &node.Status.Conditions[i]
+			}
+		}
+
+		return nil
+	}
+
 	notReadyNodes := []string{}
 	for _, node := range nodes {
-		for _, con := range node.Status.Conditions {
-			if con.Type == coreapiv1.NodeReady && con.Status != coreapiv1.ConditionTrue {
```

Comment:
it's a bit more defensive, if the "Ready" condition is not found then we should deem the node as `not ready`, correct?

Reply:
in implementation logic we have to take into account the possibility of the condition not being in the slice. In my opinion, we should be defensive and guard for nil or not found although by contract the `Nodeready` condition should be there.